src/main/java/com/iexec/sms/secret/MeasuredSecretService.java

```diff
@@ -0,0 +1,117 @@
+    private static final String STORED_SECRETS_COUNT_POSTFIX = "stored";
+
+    private final String secretsType;
+    private final String metricsPrefix;
+    private final Supplier<Long> storedSecretsCountGetter;
+    private final ScheduledExecutorService storageMetricsExecutorService;
+    private final int storedSecretsCountPeriod;
+
+    private Counter initialSecretsCounter;
+    private AtomicLong storedSecretsCount;
+    private final Counter addedSecretsSinceStartCounter;
+
+    public MeasuredSecretService(String secretsType,
+                                 String metricsPrefix,
+                                 Supplier<Long> storedSecretsCountGetter,
+                                 ScheduledExecutorService storageMetricsExecutorService,
+                                 int storedSecretsCountPeriod) {
+        this.secretsType = secretsType;
+        this.metricsPrefix = metricsPrefix;
+        this.storedSecretsCountGetter = storedSecretsCountGetter;
+        this.storageMetricsExecutorService = storageMetricsExecutorService;
+        this.storedSecretsCountPeriod = storedSecretsCountPeriod;
+
+        this.addedSecretsSinceStartCounter = Metrics.counter(metricsPrefix + ADDED_SECRETS_SINCE_START_COUNT_POSTFIX);
+    }
+
+    @PostConstruct
+    void init() {
+        final long initialSecretsCount = storedSecretsCountGetter.get();
+        this.initialSecretsCounter = Metrics.counter(metricsPrefix + INITIAL_SECRETS_COUNT_POSTFIX);
+        this.initialSecretsCounter.increment(initialSecretsCount);
+
+        this.storedSecretsCount = Metrics.gauge(metricsPrefix + STORED_SECRETS_COUNT_POSTFIX, new AtomicLong(initialSecretsCount));
+
+        storageMetricsExecutorService.scheduleAtFixedRate(
+                this::countStoredSecrets,
+                storedSecretsCountPeriod,
+                storedSecretsCountPeriod,
+                TimeUnit.SECONDS
+        );
+    }
+
+    // region Get metrics
+    public long getInitialSecretsCount() {
+        return (long) initialSecretsCounter.count();
+    }
+
+    public long getAddedSecretsSinceStartCount() {
+        return (long) addedSecretsSinceStartCounter.count();
+    }
+
+    public long getStoredSecretsCount() {
+        return storedSecretsCount.get();
+    }
+    // endregion
+
+    /**
+     * Indicate to this service a new secret has been added to the DB,
+     * so it can update its counter.
+     */
+    public void newlyAddedSecret() {
+        addedSecretsSinceStartCounter.increment();
+    }
+
+    private void countStoredSecrets() {
+        try {
+            final Long count = storedSecretsCountGetter.get();
+            log.debug("Counting secrets [type:{}, count:{}]", secretsType, count);
+            storedSecretsCount.set(count);
+        } catch (RuntimeException e) {
+            log.error("Secrets count has failed [type:{}]", secretsType, e);
+            storedSecretsCount.set(-1);
```

Comment:
IMHO, we should have a mechanism to detect when we were unable to access the database. It may be:
- setting the "stored secrets count" value to -1 ;
- setting a "can't access DB" flag to 1.

I don't think I have a preference for any of these solutions. So, if you have any, let me know and I'd be happy to implement it.